rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,316 @@
+
+- Support (REST / GRPC)
+- Implement API endpoint for cli and writer
+
+### Cli
+
+Provides some ability to operate Table Management Service, such as List all instances, Add instance, Remove instance, Clear jobs for table (etc)
+
+### Writer
+
+- Enable meta server
+    - commit instant to meta server and skip any scheduling + execution of table services
+- Do not enable meta server
+    - commit instant and request to Table Management Service for scheduling + execution of table services
+
+### Multiple instances
+
+Table Management Service instances are stateless, each instance is processed based on commit requests (scheduling + execution of table services), and non-repeated scheduling is guaranteed through meta server or ZK locking
+
+### Scheduler
+
+- Periodically scan the storage and submit operation job according to user-specified rules, like priority, queue, owner
+  etc.
+- Need to plug-in the execution engine.
+
+### Storage Schema
+
+- **tbl**
+
+    - unique key: db_name, table_name
+
+    - | name         | type   | comment               |
+            | ------------ | ------ | --------------------- |
+      | id           | bigint | auto_increment        |
+      | db_name      | string | database name         |
+      | table_name   | string | table name            |
+      | base_path    | string | storage path          |
+      | owner        | string | owner                 |
+      | queue        | string | queue                 |
+      | action_types | string | register action types |
+      | priority     | string | priority              |
+
+- **operation**
+
+    - unique key: tbl_id, instant
+
+    - | name           | type      | comment                                            | 
+      | -------------- | --------- | -------------------------------------------------- |
+      | operation_id   | bigint    | auto_increment                                     |
+      | tbl_id         | string    | table the operation belongs to                     |
+      | instant        | string    | operation timestamp                                |
+      | action         | string    | operation action, compaction, clustering and etc   |
+      | status         | int       | operation status                                   |
+      | engine         | string    | operation engine, Spark or Flink                   |
+      | retry_on_error | boolean   | operation should retry on error or not             |
+      | run_times      | int       | operation run times                                |
+      | isDeleted      | boolean   | true if this operation has been deleted in service |
+      | schedule_time  | timestamp | operation created time                             |
+      | create_time    | timestamp | operation created time                             |
+      | update_time    | timestamp | operation updated time                             |
+
+## Interface design
+
+### Register
+
+/v1/hoodie/service/register
```

Comment:
please use `hudi` instead of `hoodie`. same as explained previously.

Reply:
But I saw the embedded timeline service url is /v1/hoodie/xxx.

Reply:
ok right. let's be consistent then.